backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala

```diff
@@ -216,15 +216,26 @@ object VeloxBackendSettings extends BackendSettingsApi {
 
     // Validate if all types are supported.
     def validateDataTypes(): Option[String] = {
-      val unsupportedTypes = fields.flatMap {
-        field =>
-          field.dataType match {
-            case _: StructType => Some("StructType")
-            case _: ArrayType => Some("ArrayType")
-            case _: MapType => Some("MapType")
-            case _: YearMonthIntervalType => Some("YearMonthIntervalType")
+      val unsupportedTypes = format match {
+        case _: ParquetFileFormat =>
+          fields.flatMap {
+            case StructField(_, _: YearMonthIntervalType, _, _) =>
+              Some("YearMonthIntervalType")
+            case StructField(_, _: StructType, _, _) =>
```

Comment:
@weixiuli Velox doesn't support struct(struct), right?

Reply:
Velox already supports some struct structures, but struct structures are more complex, some UTs can not run through, therefore, this PR only supports the frequently used MAP and ARRAY, I will support struct in the later PR.

Reply:
> @weixiuli Velox doesn't support struct(struct), right?

Whether or not a StructType is a constant, it is captured by the validateDataTypes and does not support vectorized writes. The `isConstantComplexType` method is used to check if ArrayType/MapType is sconstant, and if it is, it needs to fall back because velox currently supports this scenario.